### DISCUSSION TITLE
Getting a random row

### DISCUSSION THREAD

**Discussion by DraftProducts** [prisma/prisma#9527] [Q&A] [answered]
I can't see any documentation or built in way to get a random row into a database with prisma.
Does someone have an idea ?

That how we can do something similar with Sequelize:
```js
const user = await <Model>.findOne({
  order: Sequelize.random(),
  where: { candies: { [Sequelize.Op.gt]: 5 } }
});
```
https://sequelize.org/master/class/lib/sequelize.js~Sequelize.html#instance-method-random

**Comment by ryands17** [comment, marked as answer]
@DraftProducts 👋 

This isn't supported yet. There's a similar request here for random number of rows that you can add a 👍 to https://github.com/prisma/prisma/issues/5894

**Reply by DraftProducts** [reply]
Yes, I did it yesterday, thanks.
Would there be a way to do this with prisma without it being handled natively?

**Reply by ryands17** [reply]
If you're using Postgres, you can perform the following raw query:

```ts
await prisma.$queryRaw`select * from table order by random() limit 1`
```

This will select a row at random.

**Reply by DraftProducts** [reply]
Yeah, sure `$queryRaw` thank you!

**Reply by blevine** [reply]
I wonder if you could do this using pagination arguments by including:

```
take: 1
skip:  your_random_integer_generator_function()
```

**Reply by ryands17** [reply]
You can, but if the function provides a higher skip than the number of items present then it would not return anything.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
